### ISSUE TITLE
Unnecessary newline before "(" for nullable lambda definitions.

### ISSUE THREAD

**Issue by ggajews** [pinterest/ktlint#1255]
## Expected Behavior
No error reported

## Observed Behavior
ktlint is formatting following line
`var changesListener: ((width: Double?, depth: Double?, length: Double?, area: Double?) -> Unit)? = null`
like this
```
var changesListener: (
        (
            width: Double?,
            depth: Double?,
            length: Double?,
            area: Double?
        ) -> Unit
    )? = null
```

but at the same time it reports an issue:
`xxx.kt:21:9: Unnecessary newline before "("`

## Your Environment

* Version of ktlint used: 0.42.1
* Name and version (or code for custom task) of integration used (Gradle plugin, Maven plugin, command line, custom Gradle task): `org.jlleitschuh.gradle:ktlint-gradle:10.2.0`

**Comment by paul-dingemans** [comment]
Closed by #1284

**Comment by paul-dingemans** [comment]
In this case the parameter-list-wrapping rule is a bit too aggressive. Because the max-line-length is exceeded, this rule decides to move each parameter to a separate line but this should not have been applied to the parameters of the lambda as well. So the code should have been wrapped as follows:
```
var changesListener: (
    (width: Double?, depth: Double?, length: Double?, area: Double?) -> Unit
)? = null
```
This formatting is also accepted by the default IntelliJ IDEA formatting.

It becomes more interestingly when the lambda parameter would still exceed the max-line-lenght. In that case, I suggest to not wrap the rule (this strategy is used in other rules as well). Reason for this is that the default IntelliJ IDEA formatting is not able, to format this properly. For example:
```
var changesListener: (
    ( width: Double?,
      depth: Double?,
      length: Double?,
      area: Double?
    ) -> Unit
 )? = null
 ```
 will be changed to code below by default IntelliJ IDEA:
 ```
var changesListener: (
    (
    width: Double?,
    depth: Double?,
    length: Double?,
    area: Double?
) -> Unit
)? = null
```
